pints/_transformation.py

```diff
@@ -1134,3 +1147,32 @@ def sample(self, n):
         for i, p in enumerate(ps):
             qs[i, :] = self._transform.to_search(p)
         return qs
+
+
+class UnitCubeTransformation(ScalingTransformation):
+    """
+    Maps a parameter space onto the unit (hyper)cube.
+
+    Transformations from model parameters ``p`` to search parameters ``q`` are
+    made as::
+
+        q = (p - lower) / (upper - lower)
+
+    Extends :class:`ScalingTransformation`.
+    """
+    def __init__(self, lower, upper):
+
+        # Check input
+        self._lower = pints.vector(lower)
+        self._upper = pints.vector(upper)
+        self._n_parameters = len(lower)
+        del lower, upper
```

Comment:
Do we really have to del these? We could have del scaling and translation in `ScalingTransformation` too.

Reply:
It's completely unnecessary. But I sometimes like to delete a variable after so that I can't accidentally use it later (e.g. mistakenly refer to `lower` (which the user can still change) instead of `self._lower`